lib/facter/facts/solaris/ldom.rb

```diff
@@ -5,15 +5,15 @@ module Solaris
     class Ldom
       FACT_NAME = 'ldom'
       ALIASES = %w[
-        ldom_domainchassis
-        ldom_domaincontrol
-        ldom_domainname
-        ldom_domainrole_control
-        ldom_domainrole_impl
-        ldom_domainrole_io
-        ldom_domainrole_root
-        ldom_domainrole_service
-        ldom_domainuuid
+        solaris_ldom_domainchassis
+        solaris_ldom_domaincontrol
+        solaris_ldom_domainname
+        solaris_dom_domainrole_control
+        solaris_ldom_domainrole_impl
+        solaris_dom_domainrole_ios
+        solaris_ldom_domainrole_root
+        solaris_ldom_domainrole_service
+        solaris_ldom_domainuuid
```

Comment:
Are these aliases public facing, as in could someone call `Facter.resolve("solaris_ldom_domainname")` and expect to get a value back? Or are they just internal constructs?

Reply:
I'm not too sure, I can run:

```
$ bundle exec facter solaris_ldom_domainrole_impl
LDoms
```

Would more work be needed to make the aliases public facing? I've been having a hard time wrapping my head around ALIASES

Reply:
I was thinking more of the case where someone is referencing a legacy fact by its alias on the command line (`bundle exec`) `facter ldom_domainname` or in a puppet manifest:

```puppet
notice($facts['ldom_domainname'])
```

Note facter#main ships in puppet-agent#7.x where legacy facts aren't disabled by default.

I'm thinking this fix needs to be made in the code that matches the user query (`ldom.domainrole.control`) against the various `Facter::ResolvedFact` names. My guess is facter is not escaping the `.` in the user query when converting it to a regex?